xcommon.cmake

```diff
@@ -760,9 +777,25 @@ function(XMOS_REGISTER_DEPS)
             message(STATUS "Adding dependency ${DEP_NAME}")
             include(${module_dir}/lib_build_info.cmake)
 
-            manifest_git_status(${DEP_NAME} ${DEP_VERSION})
+            manifest_git_status(${DEP_NAME} ${DEP_VERSION} manifest_str)
+
+            # Create the Depends_on column in manifest
+            if(${FULL_MANIFEST})
+                pad_string(${manifest_str} 170 manifest_str)
+                list(LENGTH LIB_DEPENDENT_MODULES DEP_COUNT)
+                foreach(DEP_MODULE ${LIB_DEPENDENT_MODULES})
+                    parse_dep_string(${DEP_MODULE} DEP_REPO DEP_VERSION DEP_NAME)
+                    math(EXPR DEP_COUNT "${DEP_COUNT} - 1")
+                    set(manifest_str "${manifest_str} ${DEP_NAME}(${DEP_VERSION})")
+                    if(NOT DEP_COUNT EQUAL 0)
+                        set(manifest_str "${manifest_str},")
+                    endif()
+                endforeach()
+            endif()
+            file(APPEND ${MANIFEST_OUT} "${manifest_str}\n")
```

Comment:
Since this chunk is identical to lines 569-582, this would be better in a macro().